ios/brave-ios/Sources/AIChat/Components/Markdown/AttributedString+Markdown.swift

```diff
@@ -149,7 +149,7 @@ extension AttributedString {
     self = result
   }
   
-  func trimmingCharacters(in characterSet: CharacterSet) -> AttributedString {
+  mutating func trimmingCharacters(in characterSet: CharacterSet) -> AttributedString {
```

Comment:
You made this mutating but are still creating a copy in the function. Call `removeSubrange` on `self` with `startRange` and `endRange` instead of creating a new attributed string with that subrange

Reply:
This also should be named `trimCharacters` and not return a new string